harambe/contrib/soup/impl.py

```diff
@@ -69,6 +69,9 @@ def tracing(self) -> Tracer:
     def url(self) -> str:
         return self._url
 
+    def set_url(self, url: str) -> None:
+        self._url = url
+
```

Comment:
thoughts on this instead? imo this the below is more pythonic
``
class Example(object):
    def __init__(self, x=None, y=None):
        self._x = x
        self._y = y

    @property
    def x(self):
        return self.x or self.defaultX()

    @x.setter
    def x(self, value):
        self._x = value

    @property
    def y(self):
        return self.y or self.defaultY()

    @y.setter
    def y(self, value):
        self._y = value
 ```